depmap-compute/depmap_compute/context.py

```diff
@@ -101,3 +112,30 @@ def walk(node, key):
         return node
 
     return walk(expr, None)
+
+
+def decode_slice_id(slice_id) -> tuple[str, str, str]:
+    """
+    Originally based on the function of the same name from vector_catalog.SliceSerializer,
+    Data Explorer 2 slice ids are a superset of the legacy slice ids.
+    Originally, slice IDs were formatted like "slice/some_dataset_id/some_feature_label/label", 
+    or "slice/some_dataset_id/some_feature_id/entity_id", where the last part of the string
+    (originally called the SliceRowType) specifies whether the feature is being identified by ID or by label. 
+
+    The DE2 slice IDs give you flexibility by letting you query samples as well using the "transpose_label" specifier.
+    When "transpose_label" is used as the last segment of the slice ID, it means we should query for samples.
+    """
```

Comment:
I tried to add some better documentation to this `decode_slice_id` function because I am often confused about which slice IDs formats are supported in which parts of the codebase. I do think there is still some work remaining to clean this up, and I think there needs to be a bit more discussion about what we want from slice IDs - especially since they're stored in the DE1 and DE2 URLs, so they're difficult to change once they're in use. 

As an example of why I find this confusing, DE1 supports the slice types "label" and "entity_id", while DE2 supports "label" and "transpose_label". It seems like someday, we may want DE2 on breadbox to be able to look up features or samples by ID as well, which would mean we'd need four "slice types":
* label
* entity_id
* transpose_label
* transpose_id (?)

I'll add a note to our Breadbox discussion list to talk about this more.